eventstore/jpa/src/test/java/org/occurrent/eventstore/jpa/JpaBlockingEventStoreTestPostgres.java

```diff
@@ -0,0 +1,157 @@
+package org.occurrent.eventstore.jpa;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.persistence.EntityManagerFactory;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.occurrent.condition.Condition;
+import org.occurrent.eventstore.jpa.batteries.CloudEventDao;
+import org.occurrent.eventstore.jpa.batteries.CloudEventDaoLog;
+import org.occurrent.eventstore.jpa.batteries.CloudEventDaoLogOperations;
+import org.occurrent.eventstore.jpa.batteries.StreamEventDaoConverter;
+import org.occurrent.eventstore.jpa.utils.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@ContextConfiguration(
+    classes = {JpaBlockingEventStoreTestPostgres.PGConfig.class, RepositoryConfig.class})
+@ExtendWith(SpringExtension.class)
+@Testcontainers
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class JpaBlockingEventStoreTestPostgres
+    extends JpaBlockingEventStoreTestBase<CloudEventDao, JPAEventStore<Long, CloudEventDao>> {
+  static final String USER_NAME = "user";
+  static final String PASSWORD = "password";
+  static final String DB = "db";
```

Comment:
This (test) file contains all of the postgres specific functionality. All other functionality is implementing by standing on the shoulders of the JPA functionality. 

The notable parts are
- table structure (see `create_table` method)
  - This is making use of postgres native `JSONB` data type. This data type is great for json structures that need to be queried/compressed 
-   `PostgresOperations`
  - This is injecting/overriding some functionality supplied by the default implementation
  - This overridden functionality allows us to query nested fields in the `jsonb` column "data"